src/Session.hs

```diff
@@ -125,6 +134,40 @@ sessionRestart session@Session{..} = do
     sessionStart session cmd setup
 
 
+performEvals :: Ghci -> Bool -> [FilePath] -> IO [Load]
+performEvals _ False _ = pure []
+performEvals ghci True reloaded = do
+  cmds <- traverse getCommands reloaded
+  fmap join $ flip traverse cmds $ \(file, cmds') ->
+    flip traverse cmds' $ \(num, cmd) -> do
+      ref <- newIORef []
+      execStream ghci (intercalate " " $ lines cmd) $ \_ resp -> modifyIORef ref (++ resp)
+      resp <- readIORef ref
+      pure $ EvalResult file (num, 1) cmd resp
+
+
+getCommands :: FilePath -> IO (FilePath, [(Int, String)])
+getCommands fp = do
+  ls <- readFile fp
+  pure (fp, splitCommands $ zip [1..] $ lines ls)
+
+splitCommands :: [(Int, String)] -> [(Int, String)]
+splitCommands [] = []
+splitCommands ((num, line) : ls)
+  | isCommand line =
+      let (cmds, xs) = span (isCommand . snd) ls
+       in (num, unlines $ fmap (drop $ length commandPrefix) $ line : fmap snd cmds) : splitCommands xs
```

Comment:
So N successive lines are treated as a single block, rather than as N successive tests? But then sent over as N lines?

Reply:
Yeah, so if we write this:

```haskell
-- > let x = 5
-- >  in x
-- 
-- > let y = 6
-- >  in y
```

this will show up as two commands. Ideally we'd like to preserve the newlines, and then dump them into ghci via `:{ multiline blocks :}`, but ghcid doesn't parse ghci's response to such things correctly. Keeping the newline structure here paves the way for getting multiline working properly in the future, though today we just glom the lines together with spaces and hope it works.